test/e2e/data/autoscaler/autoscaler.yaml

```diff
@@ -0,0 +1,184 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cluster-autoscaler-system
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler
+  namespace: cluster-autoscaler-system
+  labels:
+    app: cluster-autoscaler
+spec:
+  selector:
+    matchLabels:
+      app: cluster-autoscaler
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cluster-autoscaler
+    spec:
+      containers:
+        - image: k8s.gcr.io/autoscaling/cluster-autoscaler:v1.26.0
+          name: cluster-autoscaler
+          command:
+            - /cluster-autoscaler
+          args:
+            - --cloud-provider=clusterapi
```

Comment:
i need to understand the test cluster structure a little better, but we most likely need to be passing some kubeconfigs to the autoscaler here.

it looks like we are running the autoscaler in the management cluster, if that is the case we will need to pass in a kubeconfig for the workload cluster so it can read the pods/nodes/etc.

i think it's similar to [this configuration ](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi#autoscaler-running-in-management-cluster-using-service-account-credentials-with-separate-workload-cluster)

Reply:
That makes a lot o sense 😄 

Wondering if I should run the auto scaler in the test namespaces, so the test is fully isolated from other tests.
The only bit of information I need to do this is to understand if/how it is possible to tell the auto scaler to look at node-groups (cluster API clusters/machine deployments) in a specific namespace only

Reply:
> The only bit of information I need to do this is to understand if/how it is possible to tell the auto scaler to look at node-groups (cluster API clusters/machine deployments) in a specific namespace only

definitely possible, check out the [instructions for node autodiscovery(https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi#configuring-node-group-auto-discovery), you can do exactly that with the namespace lookup.

Reply:
i would expect the autodiscovery flag to work regardless of the topology, as long as the kubeconfig that it uses to find the cluster api resources has access to the namespace.